component/alertrules.libsonnet

```diff
@@ -103,7 +64,11 @@ local dropInfoRules =
   };
 
 local prometheus_rules(name, groups, baseURL) = kube._Object('monitoring.coreos.com/v1', 'PrometheusRule', name) {
-  metadata+: {
+  metadata: {
+    labels: {
+      name: name,
+    },
+    name: name,
```

Comment:
This seems redundant since it's identical to what you get from `kube._Object` when using `metadata+`.

Reply:
Yes, but `kube._Object` also inserts an empty `annotations` field, which is not applied with `server-side` strategy, causing ArgoCD to see the manifest as unsynced.